migrate/src/main/scala/migrate/utils/ScalafixService.scala

```diff
@@ -0,0 +1,126 @@
+
+  def fixInPlace(eval: ScalafixEvaluation): Unit =
+    if (eval.isSuccessful) {
+      val filesEvaluated = eval.getFileEvaluations.toSeq
+      filesEvaluated.foreach { oneFile =>
+        val absPath = AbsolutePath.from(oneFile.getEvaluatedFile).get
+        if (oneFile.isSuccessful) {
+          oneFile.previewPatchesAsUnifiedDiff().asScala match {
+            case None => scribe.info(s"Nothing to fix in $absPath)")
+            case Some(_) => {
+              oneFile.applyPatches()
+              scribe.info(s"Syntax fixed for $absPath)")
+            }
+          }
+        } else {
+          val errorMsg = oneFile.getErrorMessage.asScala.getOrElse("Unknown Error")
+          scribe.info(s"Failed to run scalafix with ${fixSyntaxRules.mkString(", ")} on $absPath because $errorMsg")
+        }
+      }
+    } else {
+      val errorMsg = eval.getErrorMessage.asScala.getOrElse("Unknown Error")
+      scribe.info(s"Failed to run scalafix with ${fixSyntaxRules.mkString(", ")} because $errorMsg")
+
+    }
+
+  def configure(
+    newSources: Seq[AbsolutePath],
+    newCompilerOptions: Seq[String],
+    newClasspath: Classpath,
+    newTargetRootSemantic: AbsolutePath
+  ): ScalafixService =
+    this.copy(
+      sources = newSources,
+      compilerOptions = newCompilerOptions,
+      classpath = newClasspath,
+      targetRootSemantic = newTargetRootSemantic
+    )
+
+  private def evaluate(rules: Seq[String]): Try[ScalafixEvaluation] = Try {
+    val classpathWithTargetSemantic = classpath :+ targetRootSemantic
+    val args = scalafix
+      .newArguments()
+      .withRules(rules.asJava)
+      .withPaths(sources.map(_.toNio).asJava)
+      .withClasspath(classpathWithTargetSemantic.paths.map(_.toNio).asJava)
+      .withScalacOptions(compilerOptions.asJava)
+      .withToolClasspath(toolClasspath.toUrlClassLoader(scalafixClassLoader))
+    args.evaluate()
+  }
+
+}
+
+object ScalafixService {
+  val fixSyntaxRules: Seq[String] = Seq(
+    "ProcedureSyntax",
+    "fix.scala213.ConstructorProcedureSyntax",
+    "fix.scala213.ExplicitNullaryEtaExpansion",
+    "fix.scala213.ParensAroundLambda",
+    "fix.scala213.ExplicitNonNullaryApply",
+    "fix.scala213.Any2StringAdd"
+  )
+  val addExplicitResultTypesAndImplicits: Seq[String] = Seq("InferTypes", "ExplicitImplicits")
+
+  def from(
+    sources: Seq[AbsolutePath],
+    compilerOptions: Seq[String],
+    classpath: Classpath,
+    targetRootSemantic: AbsolutePath
+  ): Try[ScalafixService] =
+    for {
+      scalafix      <- Try(Scalafix.fetchAndClassloadInstance("2.13"))
+      internalRules <- Classpath.from(BuildInfo.toolClasspath)
+      externalRules <- getClassPathforRewriteRules()
+    } yield ScalafixService(
+      scalafix,
+      sources,
+      compilerOptions,
+      classpath,
+      targetRootSemantic,
+      internalRules ++ externalRules
+    )
+
+  def notConfigured: Try[ScalafixService] = from(Nil, Nil, Classpath.empty, AbsolutePath.tempDir())
```

Comment:
Looks weird that we create an `AbsolutePath.tempDir()` that is never used. I don't think we ever need a `notConfigured` service.

Also I think that the scalafix instance and the rules should be loaded only once:
```
private lazy val scalafixInstance: Try[Scalafix] = Scalafix.fetchAndClassloadInstance("2.13")
private lazy val internalRules = Classpath.from(BuildInfo.toolClasspath)
private lazy val externalRules = getClassPathforRewriteRules()

def apply(
      sources: Seq[AbsolutePath],
    compilerOptions: Seq[String],
    classpath: Classpath,
    targetRootSemantic: AbsolutePath
): Try[ScalafixService] = 
  for {
     scalafix, internalRules, externalRules
  } yield ...
```

Reply:
you are right, it's ugly. but scalafix is load only once here

Reply:
the most important part is loaded just once. then depending of the configuration (ie the project), the unamanged sources/classpath can change.